### ISSUE TITLE
Characters disappear from the attachment file name.

### ISSUE THREAD

**Issue by nicrame** [roundcube/roundcubemail#8074]
I've found very strange issue. Some characters disappear from the attachment file name, but only when some special character is there (and only some characters do that).    

My Roundcube version is 1.4.11, hosted on CentOS 7 with Apache.  
Email server is postfix/dovecot.  
PHP 7.4.19 / 5.5.68-MariaDB  

One person that were sending some files reported, that the receiver had broken attachment. The filename was completely different, with different extension. It was Outlook 2010, and yes, attachment filename was changed into "No filename 342432.dat". The numbers were random.  

I checked what is happening, and find out that file in attachment was "6753689 VDG 1,5 1,5 Pålsbo", or was it?!  
Outlook wasn't able to read it correctly. So i made file with same name, and try to send it - there was no problem in Outlook, Thunderbird or Roundcube. 

So i checked the file before it was sent and... The filename was different. It was "6753689 VDG 1,5+1,5 Pålsbo". So what happened to "+"? Well, somehow Roundcube remove it by himself. And after that attachment file name become unreadable by Outlook.  
So there is problem with Roundcube, and with Outlook. Thunderbird did read the file name correctly, but Roundcube did not! When i made new file "6753689 VDG 1,5**+**1,5 Pålsbo.txt" and send it, inside "Sent" the attachment file name is changed into "6753689 VDG 1,51,5 Pålsbo.txt".

Example screenshots:
![Zrzut ekranu (221)](https://user-images.githubusercontent.com/5872054/119375225-dc3f3180-bcba-11eb-84fb-2be1b21aa056.png)

![Zrzut ekranu (222)](https://user-images.githubusercontent.com/5872054/119375255-e3663f80-bcba-11eb-9729-ff5a7150c6b9.png)

When i attach same files with Thunderbird/Outlook there is no problem with reading file names with Roundcube:
![Zrzut ekranu (226)](https://user-images.githubusercontent.com/5872054/119375616-4d7ee480-bcbb-11eb-89a7-5c5da24c0b6c.png)
![Zrzut ekranu (227)](https://user-images.githubusercontent.com/5872054/119375634-51126b80-bcbb-11eb-87bf-b4fae8dc3d92.png)

What i find out, that character that is making problem is "**å**" from Pålsbo (it is some kind of product name used by person that reported me the problem).

When i removed "å" and change it do anything else, problem is gone.
Also another way of "walkaround" is to change filenames encoding type to "Old" one in Roundcube settings:
![Zrzut ekranu (231)](https://user-images.githubusercontent.com/5872054/119375918-aea6b800-bcbb-11eb-8e80-e8cb50b8235f.png)

Here is comparision of how Roundcube/Outlook/Thunderbird are saving attachment names (i taked files from server itself):
Roundcube:
![Zrzut ekranu (235)](https://user-images.githubusercontent.com/5872054/119376454-3e4c6680-bcbc-11eb-94dd-92a97c47c07c.png)

Outlook:
![Zrzut ekranu (236)](https://user-images.githubusercontent.com/5872054/119376720-93887800-bcbc-11eb-8200-8c2e525e14de.png)

Thunderbird:
![Zrzut ekranu (237)](https://user-images.githubusercontent.com/5872054/119376930-d0546f00-bcbc-11eb-9a6d-46aa83cbb872.png)

PS. I checked that with 2 browsers - Firefox/Chrome with same results.

**Comment by alecpl** [comment]
Let's see if I understand correctly:
1. One issue is with disappearing plus character. I confirm the issue.
2. Outlook does not support RFC2231, therefor does not display the correct filename. Here the only option is to change the setting as you did. 
3. What's the matter with "å"? I don't get it.

**Comment by nicrame** [comment]
>     3\. What's the matter with "å"? I don't get it.

When there is "å" in filename "+" is disappearing. When there is no "å", plus stays.
Preparing email:
![Zrzut ekranu (247)](https://user-images.githubusercontent.com/5872054/119960561-60d4cd00-bfa5-11eb-9e3f-de2dc2ac63de.png)
Received email:
![Zrzut ekranu (248)](https://user-images.githubusercontent.com/5872054/119960630-72b67000-bfa5-11eb-9428-501f7ec81036.png)

**Comment by alecpl** [comment]
Right. Because without special characters the encoding is not used at all. So, the only issue here is the plus character handling. Fix will be available soon.

**Comment by alecpl** [comment]
Fixed.
